src/status_im/contexts/settings/wallet/events_test.cljs

```diff
@@ -18,3 +20,47 @@
                 (sut/rename-keypair cofx
                                     [{:key-uid      key-uid
                                       :keypair-name new-keypair-name}])))))
+
+(deftest make-seed-phrase-keypair-fully-operable
+  (let [cofx            {:db {}}
+        mnemonic        "seed phrase"
+        password        "password"
+        mnemonic-masked (security/mask-data mnemonic)
+        password-masked (security/mask-data password)
+        on-success      #(prn "success")
+        on-error        #(prn "error")
+        expected        {:fx [[:json-rpc/call
+                               [{:method     "accounts_makeSeedPhraseKeypairFullyOperable"
+                                 :params     [mnemonic (native-module/sha3 password)]
+                                 :on-success [:wallet/make-seed-phrase-keypair-fully-operable-success
+                                              on-success]
+                                 :on-error   [:wallet/make-seed-phrase-keypair-fully-operable-error
+                                              on-error]}]]]}]
+    (is (= expected
+           (sut/make-seed-phrase-keypair-fully-operable
+            cofx
+            [mnemonic-masked password-masked on-success on-error])))))
```

Comment:
Here we're using `=` instead of `match?` because there are functions refs in the data, and those function refs confuse the `match?` function.

Reply:
CC: @ilmotta 

Do you know what is the exact purpose of `match?`
I've seen most of our test will work if we use `=` :thinking:

Reply:
I think `match?` will provide better outputs when the tests fail, and it allows for using functions as schema validators too. For example, I could use `fn?` to check that something is a function. However, that feature was getting in the way of testing that the passed in `on-success` or `on-error` handler were being returned in the data.

Reply:
@ulisesmac, the explanation can be found in this merge commit https://github.com/status-im/status-mobile/commit/563f1c588d1cdac31629da009a1c894133038d2d